providers/dns.rb

```diff
@@ -0,0 +1,154 @@
+#
+# Author:: Richard Lavey (richard.lavey@calastone.com)
+# Cookbook Name:: windows
+# Provider:: dns
+#
+# Copyright:: 2015, Calastone Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# See this for info on DNSCMD
+# https://technet.microsoft.com/en-gb/library/cc772069.aspx#BKMK_10
+
+include Chef::Mixin::ShellOut
+include Windows::Helper
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+action :create do
+	if @current_resource.exists
+		needsChange = (@new_resource.record_type != @current_resource.record_type) ||
+			(@new_resource.ttl > 0 && @new_resource.ttl != @current_resource.ttl) ||
+			(@new_resource.target.is_a?(String) && @new_resource.target != @current_resource.target) ||
+			(@new_resource.target.is_a?(Array) && !(@new_resource.target - @current_resource.target).empty?)
+
+		if needsChange
+			converge_by("Changing #{ @new_resource.host_name }") do
+				updateDns
+			end
+		else
+			Chef::Log.debug("#{@new_resource.host_name} already exists - nothing to do")
+		end
+	else
+		converge_by("Creating #{ @new_resource.host_name }") do
+			updateDns
+		end
+	end
+
+end
+
+action :delete do
+	if @current_resource.exists 
+		converge_by("Deleting #{ @current_resource.host_name }") do
+			executeCommand! 'recorddelete', "#{@current_resource.record_type} /f"
+		end
+	else
+		Chef::Log.debug("#{@new_resource.host_name} does not exist - nothing to do")
+	end
+end
+
+def load_current_resource
+	# validate the new resource params : A records should be an array
+	if @new_resource.record_type == 'A' && @new_resource.target.is_a?(String)
+		raise "target property must be an array for record_type A"
+	end
+	
+	@current_resource = Chef::Resource::WindowsDns.new(@new_resource.name)
+	@current_resource.host_name(@new_resource.host_name)
+	@current_resource.dns_server(@new_resource.dns_server)
+	
+	parts = @current_resource.host_name.scan(/(\w+)\.(.*)/)
+	@host = parts[0][0]
+	@domain = parts[0][1]
+
+	getAttributes
+end
+
+private
+def getAttributes
+	@command = locate_sysnative_cmd("dnscmd.exe")
```

Comment:
Get-DnsServerResourceRecord seems to be Windows 2012 only...I need to target 2008R2